src/benchmarks/micro/libraries/Microsoft.Extensions.Primitives/StringValuesBenchmark.cs

```diff
@@ -0,0 +1,238 @@
+        public void ForEach_Array()
+        {
+            for (var i = 0; i < Iterations; i++)
+            {
+                ForEach_Array_Body();
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private string ForEach_Array_Body()
+        {
+            var s = "";
+            foreach (var item in _arrayBased)
+            {
+                s = item;
+            }
+
+            return s;
+        }
+
+        [Benchmark(OperationsPerInvoke = Iterations)]
+        public void PassByValue()
+        {
+            for (var i = 0; i < Iterations; i++)
+            {
+                PassByValue_Body();
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private string PassByValue_Body()
+        {
+            return ByValue(_stringBased);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private string ByValue(StringValues val) => val.ToString();
+
+#pragma warning disable CS0649
+        private struct HeaderReferences
+        {
+            public StringValues _CacheControl;
+            public StringValues _Connection;
+            public StringValues _Date;
+            public StringValues _KeepAlive;
+            public StringValues _Pragma;
+            public StringValues _Trailer;
+            public StringValues _TransferEncoding;
+            public StringValues _Upgrade;
+            public StringValues _Via;
+            public StringValues _Warning;
+            public StringValues _Allow;
+            public StringValues _ContentType;
+            public StringValues _ContentEncoding;
+            public StringValues _ContentLanguage;
+            public StringValues _ContentLocation;
+            public StringValues _ContentMD5;
+            public StringValues _ContentRange;
+            public StringValues _Expires;
+            public StringValues _LastModified;
+            public StringValues _AcceptRanges;
+            public StringValues _Age;
+            public StringValues _ETag;
+            public StringValues _Location;
+            public StringValues _ProxyAuthenticate;
+            public StringValues _RetryAfter;
+            public StringValues _Server;
+            public StringValues _SetCookie;
+            public StringValues _Vary;
+            public StringValues _WWWAuthenticate;
+            public StringValues _AccessControlAllowCredentials;
+            public StringValues _AccessControlAllowHeaders;
+            public StringValues _AccessControlAllowMethods;
+            public StringValues _AccessControlAllowOrigin;
+            public StringValues _AccessControlExposeHeaders;
+            public StringValues _AccessControlMaxAge;
+
+            public byte[] _rawConnection;
+            public byte[] _rawDate;
+            public byte[] _rawTransferEncoding;
+            public byte[] _rawServer;
+        }
+#pragma warning restore CS0649
```

Comment:
similarly to `ClearHeaders`, this benchmark does not belong here because it's measuring the cost of passing a big struct to a method

```suggestion
```